cli/lock.ml

```diff
@@ -570,11 +588,28 @@ let run (`Root root) (`Recurse_opam recurse) (`Build_only build_only)
       ~require_cross_compile ~preferred_versions ~ocaml_version
       ~local_opam_files:opam_files ~target_packages
   in
+  let dependency_table =
+    let aux { D.Opam.Dependency_entry.package_summary; vendored } =
+      (package_summary.package, vendored)
+    in
+    dependency_entries |> List.to_seq |> Seq.map aux |> OpamPackage.Map.of_seq
+  in
   Common.Logs.app (fun l -> l "Calculating exact pins for each of them.");
   let* duniverse = compute_duniverse ~dependency_entries >>= resolve_ref in
   let target_depexts = target_depexts opam_files target_packages in
+  let* pin_depends = root_pin_depends opam_files in
+  let local_pins = get_local_pins source_config in
+  let pin_depends =
+    List.filter
+      ~f:(fun (p, _) ->
+        match OpamPackage.Map.find_opt p dependency_table with
+        | None -> assert false
```

Comment:
In this case `find` can also be used, if that is not a case that we expect to happen.